src/metabase/notification/payload/impl/system_event.clj

```diff
@@ -297,22 +300,54 @@
 
 (mu/defmethod notification.payload/notification-payload :notification/system-event :- :map
   [notification-info :- ::notification.payload/Notification]
+  (def notification-info notification-info)
   (transform-event-info notification-info))
 
 (defmethod notification.payload/notification-payload-schema :notification/system-event
   [notification-info]
   (when-let [[op _arg-schema return-schema] (some->> notification-info
-                                                     dispatch-on-event-info
+                                                     :payload
+                                                     :event_name
                                                      (get-method transform-event-info)
                                                      meta
                                                      :schema)]
     (assert (= op :=>))
     (mr/resolve-schema return-schema)))
 
+(defn- default-row-action-transformation
+  [event-topic event-info]
+  (lib.util.match/match-one
+    event-info
+    {:row_change {:before ?before
+                  :after  ?after}
+     :args       {:table_id ?table-id}}
+    {:record     (or ?after ?before)
+     :table_id   ?table-id
+     :event_name event-topic}))
+
+(mr/def ::condition.row.mutated
+  [:map {:closed true}
+   [:record :map]
+   [:table_id pos-int?]
+   [:event_name :keyword]])
+
+(mu/defmethod event-info->condition-context :event/row.created :- ::condition.row.mutated
+  [event-topic event-info]
+  (default-row-action-transformation event-topic event-info))
+
+(mu/defmethod event-info->condition-context :event/row.updated :- ::condition.row.mutated
+  [event-topic event-info]
+  (default-row-action-transformation event-topic event-info))
+
+(mu/defmethod event-info->condition-context :event/row.deleted :- ::condition.row.mutated
+  [event-topic event-info]
+  (default-row-action-transformation event-topic event-info))
```

Comment:
Why not use `derive` on these keys so you can just define the transformation once?